src/general/Species.cs

```diff
@@ -68,6 +68,11 @@ protected Species(uint id)
     /// </summary>
     public abstract string StringCode { get; set; }
 
+    /// <summary>
+    ///   Repositions the structure of the species to the center of mass
+    /// </summary>
+    public abstract void RepositionToCenterOfMass();
```

Comment:
As anything in Species should apply to *all* stages, this should be named something like RepositionToOrigin, and it would then call the stage specific logic to make that happen right.